queries/gdscript/highlights.scm

```diff
@@ -41,16 +47,46 @@
   (name) @function (parameters
     (identifier) @parameter)*)
 
-(typed_parameter
-  (identifier) @parameter)
+(typed_parameter (identifier) @parameter)
+(default_parameter (identifier) @parameter)
+
+(call (identifier) @function.call)
+(call (identifier) @include
+      (#any-of? @include "preload" "load"))
+
+;; Properties and Methods
+
+; We'll use @property since that's the term Godot uses.
+; But, should (source (variable_statement (name))) be @property, too? Since a
+; script file is a class in gdscript.
+(class_definition
+  (body (variable_statement (name) @property)))
+
+; Same question but for methods?
+(class_definition
+  (body (function_definition (name) @method)))
 
-(default_parameter
-  (identifier) @parameter)
+(attribute_call (identifier) @method.call)
+(attribute (_) (identifier) @property)
 
-(call
-  (identifier) @function)
-(attribute_call
-  (identifier) @function)
+;; Enums
+
+(enumerator left: (identifier) @constant)
+
+;; Special Builtins
+
+((identifier) @variable.builtin
+  (#any-of? @variable.builtin "self" "super"))
+
+((attribute (identifier) (attribute_call (identifier) @constructor))
+ (#eq? @constructor "new"))
+
+;; Match Pattern
+
+[
+  (underscore)
+  (pattern_open_ending)
```

Comment:
what is pattern_open_ending? and is underscore used in a pinctuation-like manner?

Reply:
Good catch, underscore should be renamed to `pattern_wildcard` in the grammar. Here's example usage of the two:

```gdscript
match some_value:
    ["open ended array pattern", ..]: pass
    ["match this", _, "and this"]: pass
    _: print("Match anything here")
```

Reply:
Go has `_` under `@constant`, and `..` under `@operator`, does that fit better here? Seems to be used similarly

Reply:
Yeah, those sound good to me.

Reply:
Done.